mdnsd/kiface.c

```diff
@@ -275,6 +141,9 @@ kev_dispatch_msg(int fd, short event, void *bula)
 		if (iface == NULL) /* this interface isn't configured */
 			continue;
 
+		sa = (struct sockaddr *)(buf + rtm->rtm_hdrlen);
```

Comment:
If there are actually multiple messages in `buf`, presumably this should be `next + rtm->rtm_hdrlen` or you're going to pick up the wrong socket addresses from the 2nd and subsequent messages.